app/src/main/java/org/kiwix/kiwixmobile/di/modules/DatabaseModule.kt

```diff
@@ -22,31 +22,29 @@ import dagger.Module
 import dagger.Provides
 import io.objectbox.BoxStore
 import io.objectbox.kotlin.boxFor
+import org.kiwix.kiwixmobile.database.newdb.dao.FetchDownloadDao
 import org.kiwix.kiwixmobile.database.newdb.dao.HistoryDao
 import org.kiwix.kiwixmobile.database.newdb.dao.NewBookDao
 import org.kiwix.kiwixmobile.database.newdb.dao.NewBookmarksDao
-import org.kiwix.kiwixmobile.database.newdb.dao.NewDownloadDao
 import org.kiwix.kiwixmobile.database.newdb.dao.NewLanguagesDao
 import org.kiwix.kiwixmobile.database.newdb.dao.NewRecentSearchDao
 import org.kiwix.kiwixmobile.database.newdb.entities.MyObjectBox
 import javax.inject.Singleton
 
 @Module
-class DatabaseModule {
+open class DatabaseModule {
   companion object {
     var boxStore: BoxStore? = null
   }
 
+  // NOT RECOMMENDED TODO use custom runner to load TestApplication
   @Provides @Singleton fun providesBoxStore(context: Context): BoxStore {
     if (boxStore == null) {
-      boxStore = MyObjectBox.builder().androidContext(context.applicationContext).build()
+      boxStore = MyObjectBox.builder().androidContext(context).build()
     }
     return boxStore!!
```

Comment:
Shouldn't we use safe calls(.?) instead of null checks(!!) everywhere?

Reply:
`!!` isn't a null check it is `this is a last resort because this will crash the application if this is null`

As the comment above states I am unhappy with this solution and intend to replace it in the future but for the meantime it is impossible for this to be null

Reply:
Ohh. I meant not-null assertion.